packages/cli/src/commands/governance/vote.ts

```diff
@@ -20,10 +21,19 @@ export default class Vote extends BaseCommand {
 
   async run() {
     const res = this.parse(Vote)
+    const signer = res.flags.from
+    const id = res.flags.proposalID
+    const voteValue = res.flags.vote as keyof typeof VoteValue
 
+    this.kit.defaultAccount = signer
     const governance = await this.kit.contracts.getGovernance()
 
-    const tx = await governance.vote(res.flags.proposalID, res.flags.vote as keyof typeof VoteValue)
-    await displaySendTx('voteTx', tx, { from: res.flags.from })
+    await newCheckBuilder(this, signer)
+      .isVoteSignerOrAccount()
+      .proposalExists(id)
+      .proposalInStage(id, 'Referendum')
```

Comment:
don't we have an enum for proposal stages, probably we should.. and we could use it here and on the other checks

Reply:
these strings are restricted to the `keyof typeof ProposalStage` enum but I can import and use the properties as well

Reply:
the oclif input is also restricted to an enum already